tests/app/dao/test_inbound_numbers_dao.py

```diff
@@ -79,3 +79,21 @@ def test_set_inbound_number_active_flag(notify_db, notify_db_session, sample_ser
     inbound_number = dao_get_inbound_number_for_service(sample_service.id)
 
     assert inbound_number.active is active
+
+
+def test_dao_allocate_number_for_service(notify_db_session):
+    number = '078945612'
+    inbound_number = create_inbound_number(number=number)
+    service = create_service()
+
+    updated_inbound_number = dao_allocate_number_for_service(service_id=service.id, inbound_number_id=inbound_number.id)
+    assert service.get_inbound_number() == number
+    assert updated_inbound_number.service_id == service.id
+
+
+def test_dao_allocate_number_for_service(notify_db_session, sample_service):
```

Comment:
Add a test for when there is no inbound_number i.e. remove the line inbound_number = create_inbound_number(number=number)?